activerecord/lib/active_record/connection_adapters/mysql/type_metadata.rb

```diff
@@ -6,14 +6,22 @@ module MySQL
       class TypeMetadata < DelegateClass(SqlTypeMetadata) # :nodoc:
         undef to_yaml if method_defined?(:to_yaml)
 
+        include Deduplicable
+
         attr_reader :extra
 
-        def initialize(type_metadata, extra: "")
+        def initialize(type_metadata, extra: nil)
```

Comment:
The vast majority of columns won't have an `extra`, `nil` seems a much better default value as we won't have to deduplicate an empty string.